src/db/migrations/migration2.js

```diff
@@ -0,0 +1,116 @@
+const DBHelper = require('../db-helper');
+const { TX_STATUS, EVENT_STATUS } = require('../../constants');
+const MultipleResultsEventApi = require('../../api/multiple-results-event');
+const EventLeaderboard = require('../../models/event-leaderboard');
+const GlobalLeaderboard = require('../../models/global-leaderboard');
+
+const PROMISE_CONCURRENCY_LIMIT = 15;
+const limit = pLimit(PROMISE_CONCURRENCY_LIMIT);
+async function wrapper (dbMethod, callback) {
+  await dbMethod();
+  callback();
+}
+async function migration2(next) {
+  try {
+    // migrate event leaderboard
+    const events = await DBHelper.findEvent({ txStatus: TX_STATUS.SUCCESS });
+    const promises = [];
+
+    events.forEach(async (event, key) => {
+      const eventAddress = event.address;
+      const resultSet = await DBHelper.findResultSet({ eventRound: 0, txStatus: TX_STATUS.SUCCESS, eventAddress: event.address });
+      const bets = await DBHelper.findBet({
+        eventAddress: event.address,
+        txStatus: TX_STATUS.SUCCESS,
+      });
+      const txs = resultSet.concat(bets);
+
+      try {
+        let i = 0;
+        await async.whilst(
+          check => check(null, i < txs.length), // trigger iter
+          (next) => {
+            const tx = txs[i];
+            i++;
+            try {
+              const userAddress = tx.betterAddress || tx.centralizedOracleAddress;
+              const eventLeaderboardEntry = new EventLeaderboard({
+                eventAddress,
+                userAddress,
+                investments: tx.amount, // event leaderboard entry already has user's investments
+                winnings: '0',
+              });
+              wrapper(async () => {
+                  await DBHelper.insertEventLeaderboard(eventLeaderboardEntry)
+                  if(event.status === EVENT_STATUS.WITHDRAWING) {
+                    const globalLeaderboardEntry = new GlobalLeaderboard({
+                      userAddress,
+                      investments: tx.amount,
+                      winnings: '0',
+                    })
+                    await DBHelper.insertGlobalLeaderboard(globalLeaderboardEntry)
+                  }
+                },
+                ()=>next(null, i))
+            } catch (err) {
+              next(err, i); // err met, trigger the callback to end this loop
+            }
+          },
+        );
+      } catch (err) {
+        // will only be called if should end this loop
+        logger.error(err);
+        throw err;
+      }
+      if (event.status === EVENT_STATUS.WITHDRAWING) {
+        const addresses = new Set();
+        txs.forEach((tx, index) => {
+          if(tx.resultIndex !== event.currentResultIndex) return;
+          if(tx.betterAddress) addresses.add(tx.betterAddress);
+          if(tx.centralizedOracleAddress) addresses.add(tx.centralizedOracleAddress);
+        })
+        addresses.forEach((address, index) => {
+          promises.push(limit(async () => {
```

Comment:
move this `limit` call to the very outer loop. so 1 promise will take care of 1 event.

this outer loop `events.forEach(async (event, key) => {` will keep running even though the `limit` here will pause operation so it "should" queue everything up anyways once it hits the 15 concurrency limit. prefer to apply the `limit` on the entire event parsing and have each `calculateWinnings` call sequentially. so each `limit` func will correspond to all the inserts + calcWinnings of one event, and calcWinnings will only be called once at a time per event. 

1. move `limit` to very outer loop
2. use `eachOfSeries` in async lib to handle calculateWinnings